src/ResultSet.php

```diff
@@ -76,6 +76,10 @@ public function next(): void
     {
         if (! $this->isCacheDisabled) {
             $this->cache->next();
+
+            if ($this->cache->valid()) {
+                $this->position += 1;
```

Comment:
Please move this into an else case of the last condition. This avoids a second call to `cache->valid()`.